src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,55 @@
+package core.basesyntax.dao.impl;
+
+import core.basesyntax.dao.ManufacturerDao;
+import core.basesyntax.db.Storage;
+import core.basesyntax.lib.Dao;
+import core.basesyntax.model.Manufacturer;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        for (Manufacturer manufacturer : Storage.manufacturers) {
+            if (manufacturer.getId() == id) {
+                return Optional.of(manufacturer);
+            }
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        for (Manufacturer oldManufacturer : Storage.manufacturers) {
+            if (oldManufacturer.getId() == manufacturer.getId()) {
+                int index = Storage.manufacturers.indexOf(oldManufacturer);
+                Storage.manufacturers.set(index, manufacturer);
```

Comment:
you may get IndexOutOfBound here which will not be obvious for the user of your code, I suggest this approach 
IntStream.range(0, Storage.manufacturers.size())
                .filter(i -> Storage.manufacturers.get(i).getId().equals(manufacturer.getId()))
                .forEach(i -> Storage.manufacturers.set(i, manufacturer));
        return manufacturer;